modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -326,7 +339,12 @@ private void defineInvokableSymbol(BLangInvokableNode invokableNode, BInvokableS
     }
 
     private void defineInvokableSymbolParams(BLangInvokableNode invokableNode, BInvokableSymbol symbol) {
-        SymbolEnv invokableEnv = SymbolEnv.createPkgLevelSymbolEnv(invokableNode, env, symbol.scope);
+        SymbolEnv invokableEnv;
+        if (invokableNode instanceof BLangAction || invokableNode instanceof BLangResource) {
```

Comment:
Shall we pass the env as a parameter to this method? The correct env can be created in the following methods.

public void visit(BLangFunction funcNode) 
public void visit(BLangAction actionNode)
public void visit(BLangResource resourceNode)

Reply:
Fixed with 6a6642b8094d731de9e332c0f6b3a52a7b54bfcc